src/shared/watchdog.c

```diff
@@ -20,6 +21,7 @@ static char *watchdog_device;
 static usec_t watchdog_timeout; /* 0 → close device and USEC_INFINITY → don't change timeout */
 static usec_t watchdog_pretimeout; /* 0 → disable pretimeout and USEC_INFINITY → don't change pretimeout */
 static usec_t watchdog_last_ping = USEC_INFINITY;
+static bool supports_pretimeout = false; /* Depends on kernel state that might change at runtime */
```

Comment:
i'd really stick to existing prefix of the naming scheme here, and name this `watchdog_supports_pretimeout`.

Reply:
shouldn't this be an `int`, i.e. tri-state, originally set to -1, and set to 0 or 1 once we figured out things?

Reply:
> shouldn't this be an `int`, i.e. tri-state, originally set to -1, and set to 0 or 1 once we figured out things?

No, because it's not cached - see discussion about kernel mods shenanigans. It's checked every time the configuration is changed.

Reply:
hmm, why is it a global field then? I mean, if it's not "cached", then you are saying it is retrieved again on each operation? but if so, then it can be on the stack, and not a static variable?

Reply:
To clarify, it's cached while pinging (and thus deciding the ping interval in `calc_timeout()`), hence why it's global. It is not cached anymore when changing the configuration, like it was in the initial version (where it was set after checking the ioctl once and then never again for the runtime of pid1)

Reply:
but i don't see any code that would ever reset it? It's set to false when PID 1 initialize, and then there's exactly one place where we set it to true, but zero places where we set it to false. That looks wrong.

Hence "not cached anymore when changing the configuration" doesn't really make sense to me?

Reply:
Good point, I've changed it so that it's now reset every time we go through `update_pretimeout()`